sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/utils/UpdatingSpecificationFilesGenerator.java

```diff
@@ -0,0 +1,253 @@
+      content = content.replaceAll("GenericCheckTemplateTest", checkName + "Test");
+      Files.write(checkTestPath, content.getBytes(charset));
+    } catch (IOException e) {
+      LOG.error("Error while writing Check Test file with name \"" + checkName + "Test.java\", please fix manually", e);
+      throw new RuntimeException(e);
+    }
+
+    String successMessage = String.format("Successfully generated Check \"%sTest.java\"", checkName);
+    LOG.info(successMessage);
+  }
+
+  public void writeSpecificationFileDefinition() {
+    Path specificationsDirectoy = Path.of("src", "main", "resources", "org", "sonar", "plugins", "secrets", "configuration");
+    String[] extensionsToSearchFor = new String[] {"yaml"};
+    Collection<File> files = FileUtils.listFiles(new File(specificationsDirectoy.toUri()), extensionsToSearchFor, false);
+
+    List<String> listOfFileNames = files.stream().map(File::getName).sorted().collect(Collectors.toList());
+
+    Path specificationDefinitionPath = Path.of("src", "main", "java", "org", "sonar", "plugins", "secrets", "SecretsSpecificationFilesDefinition.java");
+    Path specificationDefinitionTemplatePath = Path.of("src", "test", "resources", "templates", "SecretsSpecificationFilesDefinitionTemplate.java");
+
+    try {
+      Files.copy(specificationDefinitionTemplatePath, specificationDefinitionPath, StandardCopyOption.REPLACE_EXISTING);
+
+      String content = Files.readString(specificationDefinitionPath, charset);
+      content = content.replaceAll("<REPLACE-WITH-SET-OF-FILENAMES>", generateSpecificationDefinitionSet(listOfFileNames));
+      Files.write(specificationDefinitionPath, content.getBytes(charset));
+    } catch (IOException e) {
+      LOG.error("Error while writing SecretsSpecificationFilesDefinition", e);
+      throw new RuntimeException(e);
+    }
+
+    LOG.info("Successfully generated SecretsSpecificationFilesDefinition");
+  }
+
+  private void writeUpdatedRulesDefinition(List<String> checkNames) {
+    Set<String> uniqueCheckNames = SecretsRulesDefinition.checks().stream().map(Class::getSimpleName).collect(Collectors.toSet());
+    uniqueCheckNames.addAll(checkNames);
+
+    List<String> updatedCheckNames = new ArrayList<>(uniqueCheckNames);
+
+    Collections.sort(updatedCheckNames);
+
+    Path checkTestTemplatePath = Path.of("src", "test", "resources", "templates", "SecretsRulesDefinitionTemplate.java");
+    Path rulesDefPath = Path.of("src", "main", "java", "org", "sonar", "plugins", "secrets", "SecretsRulesDefinition.java");
+    try {
+      Files.copy(checkTestTemplatePath, rulesDefPath, StandardCopyOption.REPLACE_EXISTING);
+
+      String content = Files.readString(rulesDefPath, charset);
+      content = content.replace("<REPLACE-WITH-IMPORTS-OF-ALL-CHECKS>", generateImportsFor(updatedCheckNames));
+      content = content.replaceAll("<REPLACE-WITH-LIST-OF-CHECKS>", generateChecksMethodFor(updatedCheckNames));
```

Comment:
Why `replaceAll` instead of `replace`? The first argument is not a regex, so `replace` will be faster.